torchvision/prototype/datasets/utils/_internal.py

```diff
@@ -335,3 +340,45 @@ def _make_sharded_datapipe(root: str, dataset_size: int) -> IterDataPipe:
     # dp = dp.cycle(2)
     dp = TakerDataPipe(dp, dataset_size)
     return dp
+
+
+prod = functools.partial(functools.reduce, operator.mul)
+
+
+class NumericBinaryReader:
+    def __init__(self, file: IO, *, byte_order: str = sys.byteorder) -> None:
+        self._file = file
+        self._reverse = byte_order != sys.byteorder
+
+    def _compute_params(self, dtype: torch.dtype, shape: Sequence[int]) -> Tuple[int, bool]:
+        num_bytes_per_value = (torch.finfo if dtype.is_floating_point else torch.iinfo)(dtype).bits // 8
+        num_values = prod(shape) if shape else 1
+        chunk_size = num_bytes_per_value * num_values
+        reverse = num_bytes_per_value > 1 and self._reverse
+        return chunk_size, reverse
+
+    def read(self, dtype: torch.dtype, *, shape: Sequence[int] = ()) -> torch.Tensor:
```

Comment:
Do we ever need to pass a tuple as the shape?
That's a bit unexpected IMO.

Even if one asks for a shape of (N, M), there's not guarantee that the underlying layout is adequate.

Reply:
> Do we ever need to pass a tuple as the shape?

No, but it is quite convenient. Otherwise we need to manually compute the number of bytes to be read and reshape the results afterwards. This way, this can happen automatically. 

> Even if one asks for a shape of (N, M), there's not guarantee that the underlying layout is adequate.

True, you have use the "right" format for the data at hand. But you have to do it anyway at some point:

https://github.com/pytorch/vision/blob/0c68a6d0bf34ff3e526207b3e083e2384394ea0d/torchvision/datasets/_optical_flow.py#L378-L379

vs

```python
return reader.read(torch.float32, shape=(height, width, 2)).permute((2, 0, 1))
```